### PR TITLE
Map event selector to event name

### PR DESCRIPTION
**Pull request by kasperski95**
Based on the following PR: https://github.com/software-mansion/protostar/pull/330

- map selector to name
- refine output

Screen Shot 2022-06-02 at 16 46 19

### CODE REVIEW

**Comment by kasperski95** [comment]
> Please merge the changes, I will be easier to review

@MaksymilianDemitraszek merged master

File: protostar/commands/test/starkware/forkable_starknet.py [hunk 1]
```diff
@@ -15,6 +16,13 @@ class CheatableCarriedState(CarriedState):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.pranked_contracts_map: Dict[int, int] = {}
+        self.event_selector_to_name_map: Dict[int, str] = {}
+
+    def update_event_selector_to_name_map(
```
Comment (MaksymilianDemitraszek) [inline protostar/commands/test/starkware/forkable_starknet.py:21]: Why do we copy this data? Can't we just provide property to ForkableStarknet which maps to this map? This way we would probably be compatible with new deploy syscall

**Review by MaksymilianDemitraszek — CHANGES_REQUESTED** [review]
LGTM but some logic seems unnecessary.

Also I would like to see a test case:
- deploy test contract which deploys other contract and check if all selectors are present inside the map

File: protostar/commands/test/starkware/forkable_starknet.py (hunk 1 above)
Comment (kasperski95) [reply protostar/commands/test/starkware/forkable_starknet.py:21]: I can move this logic to `ForkableStarknet` if this is what you mean. `event_selector_to_name_map` stores mapping for all contracts whereas `local_event_selector_to_name_map` only for a single contract, that's why I need to copy the map.

Why do we need `self.pranked_contracts_map` and `CheatableCarriedState` then?

Comment (MaksymilianDemitraszek) [reply protostar/commands/test/starkware/forkable_starknet.py:21]: Ok, we will have to investigate it with cairo 0.9 but for now write a test and we can merge this

**Review by MaksymilianDemitraszek — APPROVED** [review]

**Pull request merged**